BlueBrain/Search
# #314 Performances of models are not comparable across experiments or reproductions

Fixes #314.

## Description

> `dvc metrics diff` should display table with the performances before and after, including the delta between the two. Files holding the metrics should be tracked with Git and not with DVC as of now.

During this PR:
- All `dvc run` commands outputting `metrics` file are re-run. All metrics file are now specified as `metrics files` instead of `output files` (`dvc run -M metric_file` instead of `dvc_run -o metric_file`). 
From those commands, `dvc.yaml` changed:
  - `cache: false` is appearing after each metric file. [expected changes]
  - `cmd` command line is slightly changing and `deps` order is slightly different too [not expected but do not change anything]
- `.gitignore` files were updated and all `metrics files` uploaded to git. 

## How to test?
First:
```
dvc repro
dvc metrics diff
```
No table should be displayed out of this.

Secondly: (`BBS_REVISION=v0.1.2`)
1. Modify `dropout` inside `config.cfg` from `0.2` to `0.1` for example. (line 86)
2. `git diff` is going to show
```
@@ -83,7 +83,7 @@ train_corpus = "corpora.train"
 dev_corpus = "corpora.dev"
 seed = ${system.seed}
 gpu_allocator = ${system.gpu_allocator}
-dropout = 0.2
+dropout = 0.1
 accumulate_gradient = 1
 patience = 1600
 max_epochs = 0
```
3. Run same command lines as for the first experiment: (This might takes some time)
```
dvc repro
dvc metrics diff
```
This time, the last command should give a summary table:
```
Path                                     Metric            Old      New      Change
../../metrics/ner/drug.json              entity_f1-score   0.62667  0.55208  -0.07458
../../metrics/ner/drug.json              entity_precision  0.63514  0.4569   -0.17824
../../metrics/ner/drug.json              entity_recall     0.61842  0.69737  0.07895
../../metrics/ner/drug.json              token_f1-score    0.62893  0.56716  -0.06177
../../metrics/ner/drug.json              token_precision   0.65789  0.48305  -0.17484
../../metrics/ner/drug.json              token_recall      0.60241  0.68675  0.08434
../../metrics/ner/organ.json             entity_f1-score   0.54545  0.54475  -0.00071
../../metrics/ner/organ.json             entity_precision  0.42105  0.42683  0.00578
../../metrics/ner/organ.json             entity_recall     0.77419  0.75269  -0.02151
../../metrics/ner/organ.json             token_f1-score    0.58632  0.60751  0.02119
../../metrics/ner/organ.json             token_precision   0.46154  0.49171  0.03017
../../metrics/ner/organ.json             token_recall      0.80357  0.79464  -0.00893
../../metrics/ner/cell_type.json         entity_precision  0.61017  0.63303  0.02286
../../metrics/ner/cell_type.json         entity_recall     0.73469  0.70408  -0.03061
../../metrics/ner/cell_type.json         token_f1-score    0.76883  0.75493  -0.0139
../../metrics/ner/cell_type.json         token_precision   0.6758   0.70899  0.0332
../../metrics/ner/cell_type.json         token_recall      0.89157  0.80723  -0.08434
../../metrics/ner/organism.json          entity_f1-score   0.56917  0.55217  -0.017
../../metrics/ner/organism.json          entity_precision  0.48322  0.50397  0.02075
../../metrics/ner/organism.json          entity_recall     0.69231  0.61058  -0.08173
../../metrics/ner/organism.json          token_f1-score    0.67445  0.63507  -0.03938
../../metrics/ner/organism.json          token_precision   0.62097  0.62813  0.00716
../../metrics/ner/organism.json          token_recall      0.73802  0.64217  -0.09585
../../metrics/ner/disease.json           entity_f1-score   0.72262  0.71802  -0.00459
../../metrics/ner/disease.json           entity_precision  0.65974  0.66757  0.00783
../../metrics/ner/disease.json           entity_recall     0.79874  0.77673  -0.02201
../../metrics/ner/disease.json           token_f1-score    0.77891  0.77074  -0.00817
../../metrics/ner/disease.json           token_precision   0.72126  0.74483  0.02357
../../metrics/ner/disease.json           token_recall      0.84658  0.79852  -0.04806
...
```
(The results is longer). 

## Checklist

  (if it is not the case, please create an issue first).
  (if needed)
  (if needed)
  (if needed)
  (if a function is added or modified)
